pytorch_lightning/accelerators/ddp_backend.py

```diff
@@ -126,12 +127,36 @@ def _call_children_scripts(self):
 
         self.task_idx = 0
 
+        # wait for all the procs to start
+        sleep(2)
+
     def train(self):
         model = self.trainer.model
-        results = self.ddp_train(process_idx=self.task_idx, model=model, is_master=True)
-        del os.environ['WORLD_SIZE']
+        results = self.ddp_train(process_idx=self.task_idx, mp_queue=None, model=model, is_master=True)
+        if 'WORLD_SIZE' in os.environ:
+            del os.environ['WORLD_SIZE']
         return results
 
+    def training_step(self, args):
+        if self.trainer.amp_backend == AMPType.NATIVE:
+            with torch.cuda.amp.autocast():
+                output = self.trainer.model(*args)
+        else:
+            output = self.trainer.model(*args)
+        return output
+
+    def validation_step(self, args):
+        output = self.training_step(args)
+        return output
+
+    def test_step(self, args):
+        output = self.training_step(args)
+        return output
+
+    def barrier(self, name: str = None):
```

Comment:
```suggestion
    def barrier(self, name: Optional[str] = None):
```

Reply:
good suggestion. i'll change in a separate PR since this affects every accelerator